### DISCUSSION TITLE
Reasons for History report being temporarily not available and what has to be done

### DISCUSSION THREAD

**Discussion by shwarcu** [LironEr/bundlemon#52] [General] [answered]
Hello 👋 
I would like to open discussion on the History report feature. I have few questions to ask and I don't think they fit the 'issue' category :) 

- What is reason behind History report not working? I can't find this in docs, or any related issue in this repo where it would be explained. What needs to be done to have that feature working? Are there any blockers to make that work again?
- Are you, @LironEr, open to contributions from other people? Do you need some help with that feature? 

I will be testing `bundlemon` in the project I work in and that particular feature is something that interests me a lot. Ideally it would be fantastic to allow bundlemon send results to self-hosted bundle size tracker or storage. I am wondering if this is doable.

**Comment by LironEr** [comment, marked as answer]
Hey (:

When I started this project I had a private repo for the server & website, Since then I moved everything to this repo because I wanted all the code to be available to everyone, but I didn't move the history part because it had some bugs and the UX was not as good as I wanted it to be.

I should have some time on the weekend so I will add back this feature, hopefully, with fewer bugs, but the UX will still be a problem 😅

I would love contributions from other people, I will add some documentation on how to run the project locally so everybody can easily contribute.

Theoretically, bundlemon service can be self-hosted, I need to test it and write documentation about it.

I will update you on the progress later this week

**Reply by shwarcu** [reply]
Thanks a lot for explaining 😁   I will be eagerly waiting for the documentation. I am QA Engineer that does a lot around automation, if you need some help writing unit/component/integration/ tests let me know, I will try to find some time for contributions once you publish some guidelines/todos/plans 👍

**Reply by LironEr** [reply]
I Just pushed & deployed the history report.

Hopefully, I will make some improvements over the next few weeks, but it's a start (:

To view the history report just change the project id to yours
```
https://app.bundlemon.dev/projects/PROJECT_ID/reports
```

If you don't have a project, you can view BundleMon website [history report](https://app.bundlemon.dev/projects/60a928cfc1ab380009f5cc0b/reports?branch=master&resolution=days)
